arch/arc/core/reset.S

```diff
@@ -202,4 +202,20 @@ _master_core_startup:
 	jl z_arc_firq_stack_set
 #endif
 
+#ifdef __CCAC__
+    /* zero the .device_states section:*/
+    mov   r0, __device_states_start
+    mov   r1, __device_states_end
+
+    mov   r2, 0x0
+_dev_st_init:
+	/*
+	 * The section is unaligned by-default, but it is quite short.
+	 * Use single byte clearing here
+	 */
+    stb	  r2, [r0, 0]
+    add   r0, r0, 1
+    BRLT  r0, r1, _dev_st_init
+#endif
+
```

Comment:
Can we move this initialization to early C code (like we do initialization of BSS) - it will be much simpler than. I.e. to ARC _PrepC.